interop/klr/gather.c

```diff
@@ -1237,6 +1237,28 @@ bool specialize(struct kernel *k, PyObject *args, PyObject *kws) {
       if (!s)
         return false;
 
+      // Process special argument $grid
+      if(strncmp(s, "$grid", 5) == 0) {
+        if (!PyLong_Check(val)) {
+          PyErr_SetString(PyExc_ValueError, "grid must be an integer");
+          return false;
+        }
+        long grid_val = PyLong_AsLong(val);
+        if (grid_val < 0 || grid_val > 8) {
+          PyErr_SetString(PyExc_ValueError, "grid must be between 0-8");
+          return false;
+        }
+        k->grid = (uint8_t) grid_val;
+        printf("Setting NKI grid to %d\n", k->grid);
+        
+        // Remove the key from kwargs so that we don't process it further or as varargs
+        if (PyDict_DelItem(kws, key) < 0) {
+          PyErr_Format(PyExc_ValueError, "Unexpected error while processing grid");
+          return NULL;
```

Comment:
trivial: an exception is already raised, and false not NULL
```suggestion
          return false;
```